torch/utils/_pytree.py

```diff
@@ -108,32 +110,33 @@ def _is_leaf(pytree: PyTree) -> bool:
 # context: some context that is useful in unflattening the pytree
 # children_specs: specs for each child of the root Node
 # num_leaves: the number of leaves
+@dataclass
 class TreeSpec:
-    def __init__(self, typ: Any, context: Context, children_specs: List['TreeSpec']) -> None:
-        self.type = typ
-        self.context = context
-        self.children_specs = children_specs
-        self.num_leaves: int = sum([spec.num_leaves for spec in children_specs])
+    type: Any
+    context: Context
+    children_specs: List['TreeSpec']
 
-    def __repr__(self) -> str:
-        return f'TreeSpec({self.type.__name__}, {self.context}, {self.children_specs})'
+    def __post_init__(self) -> None:
+        self.num_leaves: int = sum([spec.num_leaves for spec in self.children_specs])
 
-    def __eq__(self, other: Any) -> bool:
-        result = self.type == other.type and self.context == other.context \
-            and self.children_specs == other.children_specs \
-            and self.num_leaves == other.num_leaves
-        # This should really not be necessary, but mypy errors out without it.
-        return cast(bool, result)
+    def __repr__(self, indent: int = 0) -> str:
+        repr_prefix: str = f'TreeSpec({self.type.__name__}, {self.context}, ['
+        children_specs_str: str = ''
+        if len(self.children_specs):
+            indent += len(repr_prefix)
```

Comment:
instead of indenting by all of repr_prefix, you might consider hanging a newline after [, so you only need to increase the indent by one or two. It'd look like

```
TreeSpec(blah, blah, [
  blah,
  blah,
])
```

Reply:
Sure, I will update to this format.